code/modules/surgery/organs/subtypes/skrell.dm

```diff
@@ -9,42 +9,34 @@
 	icon = 'icons/obj/species_organs/skrell.dmi'
 	icon_state = "skrell_headpocket"
 	origin_tech = "biotech=2"
-	w_class = WEIGHT_CLASS_TINY
+	w_class = WEIGHT_CLASS_SMALL
 	parent_organ = "head"
 	slot = "headpocket"
 	actions_types = list(/datum/action/item_action/organ_action/toggle)
-	var/obj/item/storage/internal/pocket
-
-/obj/item/organ/internal/headpocket/New()
-	pocket = new /obj/item/storage/internal(src)
-	pocket.storage_slots = 1
-	// Fit only pocket sized items
-	pocket.max_w_class = WEIGHT_CLASS_SMALL
-	pocket.max_combined_w_class = 2
-	..()
+	var/obj/item/held_item
 
 /obj/item/organ/internal/headpocket/Destroy()
-	QDEL_NULL(pocket)
-	return ..()
-
-/obj/item/organ/internal/headpocket/insert(mob/living/carbon/M, special = FALSE, dont_remove_slot = 0)
-	. = ..()
-	pocket.master_item = M // So headpockets work with adjacency checks
-
-/obj/item/organ/internal/headpocket/remove(mob/living/carbon/M, special = 0)
-	pocket.master_item = src
+	QDEL_NULL(held_item)
 	return ..()
 
 /obj/item/organ/internal/headpocket/on_life()
 	..()
 	var/obj/item/organ/external/head/head = owner.get_organ("head")
-	if(pocket.contents.len && !findtextEx(head.h_style, "Tentacles"))
-		owner.visible_message("<span class='notice'>Something falls from [owner]'s head!</span>",
-													"<span class='notice'>Something falls from your head!</span>")
+	if(held_item && !findtextEx(head.h_style, "Tentacles"))
+		owner.visible_message("<span class='notice'>[held_item] falls from [owner]'s [src]!</span>", "<span class='notice'>[held_item] falls from your [src]!</span>")
 		empty_contents()
 
 /obj/item/organ/internal/headpocket/ui_action_click()
-	pocket.MouseDrop(owner)
+	if(held_item)
+		owner.visible_message("<span class='notice'>[owner] removes [held_item] from [owner.p_their()] [src].</span>", "<span class='notice'>You remove [held_item] from your [src].</span>")
+		owner.put_in_hands(held_item)
+		held_item = null
+	else
+		var/obj/item/I = owner.get_active_hand()
+		if(I && I.w_class <= WEIGHT_CLASS_SMALL && !istype(I, /obj/item/disk/nuclear) && owner.unEquip(I))
```

Comment:
And to echo what TDSSS said, why exclude the nuke disk from being held in headpockets? It functions exactly the same as a regular pocket and can be easily emptied:
![image](https://user-images.githubusercontent.com/57483089/109995817-8b94e980-7d06-11eb-940a-850b263681e9.png)

Reply:
As, to my knowledge, the only current user of this specific mechanic -due to playing skrell as well as captain-, I can't help but second what Sabre said, putting the nuke disk in the headpocket is of little impact, anyone who pays attention to your specie is likely to find out the trick (especially considering you're already toast if the nukies manage to get their hands on you), on top of that, you have no way of hiding this pocket unless wearing a hardsuit helmet.

The reason why I do it is that it's a very easy way to check on the disk and quickly cope with the anxiety I get from playing captain on war ops, since you only have one item, in an easy to access spot, while it may to some extent "hide" the disk, it's no more than just slapping it in a cardboard box in your bag.

Reply:
The very nature of how organs are handled now is that they're in nullspace---ergo anything you hold onto is also in null space, even the item has a `loc` inside the head pocket.

The pinpointer won't work at all as it won't be able to find it.

Refactor a ton of code just so one species can hold the nuke disk in their special organ? Not something worth it, since it functions 99.99% of the time for other things.